peas/weather.py

```diff
@@ -784,175 +740,55 @@ def calculate_and_set_PWM(self):
                 ))
                 self.set_PWM(new_PWM)
 
-    def make_safety_decision(self, current_values):
-        """
-        Method makes decision whether conditions are safe or unsafe.
-        """
-        self.logger.debug('Making safety decision')
-        self.logger.debug('Found {} weather data entries in last {:.0f} minutes'.format(
-            len(self.weather_entries), self.safety_delay))
-
-        safe = False
-
-        # Tuple with condition,safety
-        cloud = self._get_cloud_safety(current_values)
-
-        try:
-            wind, gust = self._get_wind_safety(current_values)
-        except Exception as e:
-            self.logger.warning('Problem getting wind safety: {}'.format(e))
-            wind = ['N/A']
-            gust = ['N/A']
-
-        rain = self._get_rain_safety(current_values)
-
-        safe = cloud[1] & wind[1] & gust[1] & rain[1]
-        self.logger.debug('Weather Safe: {}'.format(safe))
-
-        return {'Safe': safe,
-                'Sky': cloud[0],
-                'Wind': wind[0],
-                'Gust': gust[0],
-                'Rain': rain[0]}
-
     def _get_cloud_safety(self, current_values):
-        safety_delay = self.safety_delay
-
-        entries = self.weather_entries
-        threshold_cloudy = self.cfg.get('threshold_cloudy', -22.5)
-        threshold_very_cloudy = self.cfg.get('threshold_very_cloudy', -15.)
-
         sky_diff = [x['sky_temp_C'] - x['ambient_temp_C']
-                    for x in entries
+                    for x in self.weather_entries
                     if ('ambient_temp_C' and 'sky_temp_C') in x.keys()]
+        max_sky_diff = max(sky_diff)
+        last_cloud = current_values['sky_temp_C'] - current_values['ambient_temp_C']
 
-        if len(sky_diff) == 0:
-            self.logger.debug('  UNSAFE: no sky temperatures found')
-            sky_safe = False
-            cloud_condition = 'Unknown'
-        else:
-            if max(sky_diff) > threshold_cloudy:
-                self.logger.debug('UNSAFE: Cloudy in last {} min. Max sky diff {:.1f} C'.format(
-                                  safety_delay, max(sky_diff)))
-                sky_safe = False
-            else:
-                sky_safe = True
-
-            last_cloud = current_values['sky_temp_C'] - current_values['ambient_temp_C']
-            if last_cloud > threshold_very_cloudy:
-                cloud_condition = 'Very Cloudy'
-            elif last_cloud > threshold_cloudy:
-                cloud_condition = 'Cloudy'
-            else:
-                cloud_condition = 'Clear'
-            self.logger.debug('Cloud Condition: {} (Sky-Amb={:.1f} C)'.format(cloud_condition, sky_diff[-1]))
-
-        return cloud_condition, sky_safe
+        return super()._get_cloud_safety(max_sky_diff, last_cloud)
 
     def _get_wind_safety(self, current_values):
```

Comment:
Modify to use results from `_get_status()`, consider splitting into separate average wind speed and wind gust methods.

Reply:
Modified to use the results from `_get_status()` and I have split the wind and gust into different methods in the base class.

Reply:
With the two methods I have to somehow use them to get the values for wind and gust into weather entries so that they can be used in the base methods. Any pointers?